packages/actions/src/helpers/firebase.ts

```diff
@@ -0,0 +1,80 @@
+ * @param options <FirebaseOptions> - an object w/ every necessary Firebase option to init app.
+ * @returns <FirebaseApp> - the initialized Firebase app object.
+ */
+const initializeFirebaseApp = (options: FirebaseOptions): FirebaseApp => initializeApp(options)
+
+/**
+ * This method returns the Firestore database instance associated to the given Firebase application.
+ * @param app <FirebaseApp> - the Firebase application.
+ * @returns <Firestore> - the Firebase Firestore associated to the application.
+ */
+const getFirestoreDatabase = (app: FirebaseApp): Firestore => getFirestore(app)
+
+/**
+ * This method returns the Cloud Functions instance associated to the given Firebase application.
+ * @param app <FirebaseApp> - the Firebase application.
+ * @returns <Functions> - the Cloud Functions associated to the application.
+ */
+const getFirebaseFunctions = (app: FirebaseApp): Functions => getFunctions(app)
+
+/**
+ * Return the core Firebase services instances (App, Database, Functions).
+ * @param apiKey <string> - the API key specified in the application config.
+ * @param authDomain <string> - the authDomain string specified in the application config.
+ * @param projectId <string> - the projectId specified in the application config.
+ * @param messagingSenderId <string> - the messagingSenderId specified in the application config.
+ * @param appId <string> - the appId specified in the application config.
+ * @returns <Promise<FirebaseServices>>
+ */
+export const initializeFirebaseCoreServices = async (
+    apiKey: string,
+    authDomain: string,
+    projectId: string,
+    messagingSenderId: string,
+    appId: string
+): Promise<FirebaseServices> => {
+    const firebaseApp = initializeFirebaseApp({
+        apiKey,
+        authDomain,
+        projectId,
+        messagingSenderId,
+        appId
+    })
+    const firestoreDatabase = getFirestoreDatabase(firebaseApp)
+    const firebaseFunctions = getFirebaseFunctions(firebaseApp)
+
+    return {
+        firebaseApp,
+        firestoreDatabase,
+        firebaseFunctions
+    }
+}
+
+/**
+ * Sign in w/ OAuth 2.0 token.
+ * @param firebaseApp <FirebaseApp> - the configured instance of the Firebase App in use.
+ * @param credentials <OAuthCredential> - the OAuth credential generated from token exchange.
+ */
+export const signInToFirebaseWithCredentials = async (firebaseApp: FirebaseApp, credentials: OAuthCredential) =>
+    signInWithCredential(initializeAuth(firebaseApp), credentials)
+
+/**
+ * Return the current authenticated user in the given Firebase Application.
+ * @param firebaseApp <FirebaseApp> - the configured instance of the Firebase App in use.
+ * @returns
+ */
+export const getCurrentFirebaseAuthUser = (firebaseApp: FirebaseApp): User => {
+    const user = getAuth(firebaseApp).currentUser
+
+    if (!user) throw new Error(`Authenticated user not found for current Firebase application`)
```

Comment:
should we keep using error codes? e.g. Firebase-0001 - Authenticated ...

Reply:
Not anymore. We are going to make agnostic and method only related errors in the `actions` package. These errors can be catched in the client/frontend and displayed to the user.